plugins/device/plugin.go

```diff
@@ -28,7 +28,8 @@ func (p *PluginDevice) GRPCServer(broker *plugin.GRPCBroker, s *grpc.Server) err
 
 func (p *PluginDevice) GRPCClient(ctx context.Context, broker *plugin.GRPCBroker, c *grpc.ClientConn) (interface{}, error) {
 	return &devicePluginClient{
-		client: proto.NewDevicePluginClient(c),
+		doneCtx: ctx,
```

Comment:
Ok so `doneCtx` is closed if/when the client connection closes?

Out of scope for this PR I think, but I think we might need to discuss how (or if we still want) to implement the Shutdown RPC in the base plugin. As it has implications here I think